build-resources.gradle

```diff
@@ -6,8 +6,71 @@
 ext.coreProjects = [
         project(':data-prepper-api'),
         project(':data-prepper-core'),
-        project('data-prepper-plugins:common'),
-        project('data-prepper-expression'),
+        project(':data-prepper-plugins:common'),
+        project(':data-prepper-expression'),
         project(':data-prepper-logstash-configuration')
 ]
-ext.mavenArtifactProjects = [project(':data-prepper-api')]
+ext.mavenArtifactProjects = [
```

Comment:
Can we just publish all jars and thus remove this entirely?

Reply:
Yes, I have updated the PR by changing the core projects to reflect all the top level projects we want to publish. Then I've added a release task that recursively goes through the core projects and publish them.